exchanges/huobi/huobi_wrapper.go

```diff
@@ -1741,45 +1743,41 @@ func (h *HUOBI) FormatExchangeKlineInterval(in kline.Interval) string {
 }
 
 // GetHistoricCandles returns candles between a time period for a set time interval
-func (h *HUOBI) GetHistoricCandles(ctx context.Context, pair currency.Pair, a asset.Item, start, end time.Time, interval kline.Interval) (kline.Item, error) {
-	if err := h.ValidateKline(pair, a, interval); err != nil {
-		return kline.Item{}, err
-	}
-	klineParams := &KlinesRequestParams{
-		Period: h.FormatExchangeKlineInterval(interval),
-		Symbol: pair,
-	}
-	candles, err := h.GetSpotKline(ctx, klineParams)
+func (h *HUOBI) GetHistoricCandles(ctx context.Context, pair currency.Pair, a asset.Item, interval kline.Interval, start, end time.Time) (*kline.Item, error) {
+	req, err := h.GetKlineRequest(pair, a, interval, start, end)
 	if err != nil {
-		return kline.Item{}, err
+		return nil, err
 	}
-	ret := kline.Item{
-		Exchange: h.Name,
-		Pair:     pair,
-		Asset:    a,
-		Interval: interval,
+
+	candles, err := h.GetSpotKline(ctx, KlinesRequestParams{
```

Comment:
Please add asset check and a todo, huobi supports spot, futures, coinmargined futures but this implementation is only spot